packages/firebase_app_installations/firebase_app_installations/android/src/main/java/io/flutter/plugins/firebase/installations/firebase_app_installations/FirebaseInstallationsPlugin.java

```diff
@@ -145,6 +164,85 @@ private Task<Void> deleteId(Map<String, Object> arguments) {
     return taskCompletionSource.getTask();
   }
 
+  // Pigeon FirebaseAppInstallationsHostApi implementation.
+
+  @Override
+  public void initializeApp(
+      @NonNull AppInstallationsPigeonFirebaseApp app,
+      @NonNull AppInstallationsPigeonSettings settings,
+      @NonNull GeneratedAndroidFirebaseAppInstallations.VoidResult result) {
+    // Currently there is no per-app configurable behavior required on Android for these settings.
+    // We execute asynchronously to keep the threading model consistent.
+    cachedThreadPool.execute(
+        () -> {
+          try {
+            // Touch the instance to ensure it's initialized.
+            getInstallations(app);
+            result.success();
+          } catch (Exception e) {
+            result.error(e);
+          }
+        });
+  }
+
+  @Override
+  public void delete(
+      @NonNull AppInstallationsPigeonFirebaseApp app,
+      @NonNull GeneratedAndroidFirebaseAppInstallations.VoidResult result) {
+    cachedThreadPool.execute(
+        () -> {
+          try {
+            Tasks.await(getInstallations(app).delete());
+            result.success();
+          } catch (Exception e) {
+            result.error(e);
+          }
+        });
+  }
+
+  @Override
+  public void getId(
+      @NonNull AppInstallationsPigeonFirebaseApp app,
+      @NonNull GeneratedAndroidFirebaseAppInstallations.Result<String> result) {
+    cachedThreadPool.execute(
+        () -> {
+          try {
+            String id = Tasks.await(getInstallations(app).getId());
+            result.success(id);
+          } catch (Exception e) {
+            result.error(e);
+          }
+        });
+  }
+
+  @Override
+  public void getToken(
+      @NonNull AppInstallationsPigeonFirebaseApp app,
+      @NonNull Boolean forceRefresh,
+      @NonNull GeneratedAndroidFirebaseAppInstallations.Result<String> result) {
+    cachedThreadPool.execute(
+        () -> {
+          try {
+            FirebaseInstallations firebaseInstallations = getInstallations(app);
+            InstallationTokenResult tokenResult =
+                Tasks.await(firebaseInstallations.getToken(forceRefresh));
+            result.success(tokenResult.getToken());
+          } catch (Exception e) {
+            result.error(e);
```

Comment:
Also, I think it's best to return the error in this format rather than the raw exception.
```java
     Exception exception = task.getException();
            result.error(
                "firebase_app_installations",
                exception != null ? exception.getMessage() : null,
                getExceptionDetails(exception));
```